server/gfmd/user.c

```diff
@@ -419,6 +355,154 @@ user_enter_auth_id(
 	return (GFARM_ERR_NO_ERROR);
 }
 
+static gfarm_error_t
+user_auth_id_modify_internal(struct user *u,
+		char *auth_id_type, char *auth_user_id,
+		bool *need_to_update_dbp, bool *need_to_addp)
+{
+	enum auth_user_id_type auth_user_id_type = 0;
+	char *old_auth_user_id = NULL;
+	struct gfarm_hash_entry *entry;
+	gfarm_error_t e;
+
+	*need_to_update_dbp = false;
+
+	if ((e = gfarm_auth_user_id_type_from_name(
+		(char *)auth_id_type,
+		&auth_user_id_type)) != GFARM_ERR_NO_ERROR) {
+		return (e);
+	}
+
+	old_auth_user_id = u->auth_user_id[auth_user_id_type];
+
+	if (old_auth_user_id != NULL) {
+		if (strcmp(u->auth_user_id[auth_user_id_type],
+				auth_user_id) == 0) {
+			return (GFARM_ERR_NO_ERROR);
+		}
+
+		struct user_auth_key key = {
+			auth_user_id_type,
+			u->auth_user_id[auth_user_id_type]
+		};
+
+		entry = gfarm_hash_lookup(user_id_map, &key, sizeof(key));
+		if (entry != NULL) {
+			gfarm_hash_purge(user_id_map, &key, sizeof(key));
+		} else {
+			return (GFARM_ERR_ALREADY_EXISTS);
+		}
+		*need_to_addp = false;
+	} else {
+		*need_to_addp = true;
+	}
```

Comment:
This is OK for this code, but in general,
```
if (condition) {
  .. short clause ...
} else {
  ... long clause ...
}
```
is better than
```
if (condition) {
  ... long clause ...
} else {
  .. short clause ...
}
```
because the former is more friendly to human short-term memory.